enhancements/pipeline-tooling.md

```diff
@@ -0,0 +1,267 @@
+7. If the production index image changes before it merges, the PR drops from the queue (GOTO 4)
+8. Once merged, build a new production index image
+
+An configuration file will inform a build of the bundles to add or remove, if any, from an index:
+
+```yaml
+from: image://quay.io/operator-framework/community-operators:latest
+included:
+- ref: quay.io/coreos/something-bundle:v1
+  package: something
+  channel: stable
+- ref: quay.io/coreos/something-bundle@sha256:abcdef...
+  package: something
+  channel: alpha
+- ref: quay.io/redhat/something-else-bundle:v5.5
+  package: something-else
+  channel: v5.Y
+exluded:
+- ref: quay.io/redhat/something-else-bundle:v5.4
+  package: something-else
+  channel: v5.Y
+```
+
+A new `-f` option will be added to the `opm index add` subcommand which specifies the path of an index config file. The command will parse the given file and idempotently:
+
+- add bundles referenced by the `included` directive to the index specified by `from`
+- remove bundles referenced by the `excluded` directive from the index specified by `from`
+
+The command will also be capable of sourcing "from indexes" in varying formats using reference prefixes:
+
+- "image://" for container image references
+- "file://" for database file paths
+
+TODO(njhale): staged migration to bundle images
+
+1. create an index configuration specifying the existing catalog image in `from`
+2. reduce the number of manifest versions kept for each operator to the head of each package/channel
+3. walk manifest directories adding to the index database
+4. run `opm index add` using the index config as an overlay
+
+### Migrate away from appregistry
+
+Detailed steps about migration from the pipeline perspective can be found here: https://docs.engineering.redhat.com/display/DELIVERY/New+OCP+Application+Registry%3A+Requirements
+
+### Catalogs for minor OCP versions
+
+As an operator registry maintainer, I need a way to specify what OCP versions are supported by the operators in the registry.
+
+#### Design Proposal
+
+We can use names of branches/tags on dist-git repos as convention for what ocp versions are supported. We can then update each OCP versioned index for pushes to each branch/tag. If this is not possible we can include this in a config file similar to `art.yaml` for openshift operators.
+
+### Backing up the index image
+
+As a pipeline admin I want to recover/rebuild an index that has been lost or corrupted.
+
+#### Proposal
+
+On every update to an index, run `opm index extract` on that newly created index and push that content (list of images, files, or dbs) to a git repo as a backup.
+
+Q: Should we backup the db file, the image references, or the manifest/metadata folders?
+
+Note: Order currently matters for insert into the db so backing up the db file may be the best first option - a list of bundles will work after some index changes involving adhering to semver for operator upgrade paths.
```

Comment:
We could still back up using some kind of `manifest` command as long as the manifest has some kind of ordering concept.